In [43]:
import pandas as pd
import numpy as np
import os
pd.set_option('display.max_columns',120)

In [44]:
!ls data

2012-18_officialBoxScore.csv  metadata_officialBoxScore.pdf
2012-18_playerBoxScore.csv    metadata_playerBoxScore.pdf
2012-18_standings.csv         metadata_standing.pdf
2012-18_teamBoxScore.csv      metadata_teamBoxScore.pdf


In [45]:
official = pd.read_csv('data/2012-18_officialBoxScore.csv')
official.shape

(44284, 119)

In [46]:
official.head(1)

gmDate gmTime  seasTyp    offLNm offFNm teamAbbr teamConf    teamDiv  \
0  2012-10-30  19:00  Regular  Brothers   Tony      WAS     East  Southeast   

  teamLoc teamRslt  teamMin  teamDayOff  teamPTS  teamAST  teamTO  teamSTL  \
0    Away     Loss      240           0       84       26      13       11   

   teamBLK  teamPF  teamFGA  teamFGM  teamFG%  team2PA  team2PM  team2P%  \
0       10      19       90       32   0.3556       58       24   0.4138   

   team3PA  team3PM  team3P%  teamFTA  teamFTM  teamFT%  teamORB  teamDRB  \
0       32        8     0.25       20       12      0.6       18       21   

   teamTRB  teamPTS1  teamPTS2  teamPTS3  teamPTS4  teamPTS5  teamPTS6  \
0       39        24        15        23        22         0         0   

   teamPTS7  teamPTS8  teamTREB%  teamASST%  teamTS%  teamEFG%  teamOREB%  \
0         0         0    41.9355      81.25   0.4251       0.4    33.3333   

   teamDREB%  teamTO%  teamSTL%  teamBLK%  teamBLKR  teamPPS  teamFIC  \
0    53.8462  11.6279   12.3678   11.2434   17.2414   0.9333    67.25   

   teamFIC40  teamOrtg  teamDrtg  teamEDiff  teamPlay%   teamAR  teamAST/TO  \
0    56.0417   94.4447  105.6882   -11.2435     0.3765  18.8679         2.0   

   teamSTL/TO opptAbbr opptConf  opptDiv opptLoc opptRslt  opptMin  \
0     84.6154      CLE     East  Central    Home      Win      240   

   opptDayOff  opptPTS  opptAST  opptTO  opptSTL  opptBLK  opptPF  opptFGA  \
0           0       94       22      21        7        5      21       79   

   opptFGM  opptFG%  oppt2PA  oppt2PM  oppt2P%  oppt3PA  oppt3PM  oppt3P%  \
0       36   0.4557       59       29   0.4915       20        7     0.35   

   opptFTA  opptFTM  opptFT%  opptORB  opptDRB  opptTRB  opptPTS1  opptPTS2  \
0       22       15   0.6818       18       36       54        31        19   

   opptPTS3  opptPTS4  opptPTS5  opptPTS6  opptPTS7  opptPTS8  opptTREB%  \
0        24        20         0         0         0         0    58.0645   

   opptASST%  opptTS%  opptEFG%  opptOREB%  opptDREB%  opptTO%  opptSTL%  \
0    61.1111     0.53       0.5    33.3333    66.6667  19.1466    7.8704   

   opptBLK%  opptBLKR  opptPPS  opptFIC  opptFIC40  opptOrtg  opptDrtg  \
0    5.6217    8.4746   1.1899     74.0    61.6667  105.6882   94.4447   

   opptEDiff  opptPlay%   opptAR  opptAST/TO  opptSTL/TO     poss     pace  
0    11.2435      0.439  16.7072      1.0476     33.3333  88.9409  88.9409

#### summary of what this dataset provides

1. Game date, time ,season type, home/away, win/loss, 
2. officials name
3. team, team ids, team divison,  rating stats.
4. general playing. stats. 
5. points in. that quarter
6. more advanced stats in ercentage. 
7. same stats for oponenet too

Stats of 44,284 games between 2012-2106

**no null values** data is clean

Ideas on how to augment data/process it:
1. turn date into datetime objects, and same with time.
2. drop seasType, since its all the samne value "Regular"
3. boolean teamConf(east,west), teamLoc(away, homme),...boolean all binary columns
4. teamDiv to categorical, only 6 uniaue values

**how can this help our data?**
can merge on gmDate, officials, adn team

this can tell us how a player did in a game against another player. 
seems like this is gemnrally stats to merge for another table w/ players wince we are focusining on the other team

In [47]:
official.head()

gmDate gmTime  seasTyp    offLNm    offFNm teamAbbr teamConf  \
0  2012-10-30  19:00  Regular  Brothers      Tony      WAS     East   
1  2012-10-30  19:00  Regular     Smith   Michael      WAS     East   
2  2012-10-30  19:00  Regular   Workman  Haywoode      WAS     East   
3  2012-10-30  19:00  Regular  Brothers      Tony      CLE     East   
4  2012-10-30  19:00  Regular     Smith   Michael      CLE     East   

     teamDiv teamLoc teamRslt  teamMin  teamDayOff  teamPTS  teamAST  teamTO  \
0  Southeast    Away     Loss      240           0       84       26      13   
1  Southeast    Away     Loss      240           0       84       26      13   
2  Southeast    Away     Loss      240           0       84       26      13   
3    Central    Home      Win      240           0       94       22      21   
4    Central    Home      Win      240           0       94       22      21   

   teamSTL  teamBLK  teamPF  teamFGA  teamFGM  teamFG%  team2PA  team2PM  \
0       11       10      19       90       32   0.3556       58       24   
1       11       10      19       90       32   0.3556       58       24   
2       11       10      19       90       32   0.3556       58       24   
3        7        5      21       79       36   0.4557       59       29   
4        7        5      21       79       36   0.4557       59       29   

   team2P%  team3PA  team3PM  team3P%  teamFTA  teamFTM  teamFT%  teamORB  \
0   0.4138       32        8     0.25       20       12   0.6000       18   
1   0.4138       32        8     0.25       20       12   0.6000       18   
2   0.4138       32        8     0.25       20       12   0.6000       18   
3   0.4915       20        7     0.35       22       15   0.6818       18   
4   0.4915       20        7     0.35       22       15   0.6818       18   

   teamDRB  teamTRB  teamPTS1  teamPTS2  teamPTS3  teamPTS4  teamPTS5  \
0       21       39        24        15        23        22         0   
1       21       39        24        15        23        22         0   
2       21       39        24        15        23        22         0   
3       36       54        31        19        24        20         0   
4       36       54        31        19        24        20         0   

   teamPTS6  teamPTS7  teamPTS8  teamTREB%  teamASST%  teamTS%  teamEFG%  \
0         0         0         0    41.9355    81.2500   0.4251       0.4   
1         0         0         0    41.9355    81.2500   0.4251       0.4   
2         0         0         0    41.9355    81.2500   0.4251       0.4   
3         0         0         0    58.0645    61.1111   0.5300       0.5   
4         0         0         0    58.0645    61.1111   0.5300       0.5   

   teamOREB%  teamDREB%  teamTO%  teamSTL%  teamBLK%  teamBLKR  teamPPS  \
0    33.3333    53.8462  11.6279   12.3678   11.2434   17.2414   0.9333   
1    33.3333    53.8462  11.6279   12.3678   11.2434   17.2414   0.9333   
2    33.3333    53.8462  11.6279   12.3678   11.2434   17.2414   0.9333   
3    46.1538    66.6667  19.1466    7.8704    5.6217    8.4746   1.1899   
4    46.1538    66.6667  19.1466    7.8704    5.6217    8.4746   1.1899   

   teamFIC  teamFIC40  teamOrtg  teamDrtg  teamEDiff  teamPlay%   teamAR  \
0    67.25    56.0417   94.4447  105.6882   -11.2435     0.3765  18.8679   
1    67.25    56.0417   94.4447  105.6882   -11.2435     0.3765  18.8679   
2    67.25    56.0417   94.4447  105.6882   -11.2435     0.3765  18.8679   
3    74.00    61.6667  105.6882   94.4447    11.2435     0.4390  16.7072   
4    74.00    61.6667  105.6882   94.4447    11.2435     0.4390  16.7072   

   teamAST/TO  teamSTL/TO opptAbbr opptConf    opptDiv opptLoc opptRslt  \
0      2.0000     84.6154      CLE     East    Central    Home      Win   
1      2.0000     84.6154      CLE     East    Central    Home      Win   
2      2.0000     84.6154      CLE     East    Central    Home      Win   
3      1.0476     33.3333      WAS     East  Southeast    Away     Loss   


In [48]:
#official.gmDate = pd.to_datetime(official.gmDate)
#official.gmTime = pd.to_timedelta(official.gmTime.map(lambda x: x + ':00'))
official.teamLoc = official.teamLoc.map(lambda x: 1 if x=='Home' else 0)
official = official.drop(['seasTyp'],axis=1)
official['teamAbbr'] = official['teamAbbr'].astype('category')
official.teamRslt = official.teamRslt.map(lambda x: 1 if x=='Win' else 0)

In [49]:
official.head()

gmDate gmTime    offLNm    offFNm teamAbbr teamConf    teamDiv  \
0  2012-10-30  19:00  Brothers      Tony      WAS     East  Southeast   
1  2012-10-30  19:00     Smith   Michael      WAS     East  Southeast   
2  2012-10-30  19:00   Workman  Haywoode      WAS     East  Southeast   
3  2012-10-30  19:00  Brothers      Tony      CLE     East    Central   
4  2012-10-30  19:00     Smith   Michael      CLE     East    Central   

   teamLoc  teamRslt  teamMin  teamDayOff  teamPTS  teamAST  teamTO  teamSTL  \
0        0         0      240           0       84       26      13       11   
1        0         0      240           0       84       26      13       11   
2        0         0      240           0       84       26      13       11   
3        1         1      240           0       94       22      21        7   
4        1         1      240           0       94       22      21        7   

   teamBLK  teamPF  teamFGA  teamFGM  teamFG%  team2PA  team2PM  team2P%  \
0       10      19       90       32   0.3556       58       24   0.4138   
1       10      19       90       32   0.3556       58       24   0.4138   
2       10      19       90       32   0.3556       58       24   0.4138   
3        5      21       79       36   0.4557       59       29   0.4915   
4        5      21       79       36   0.4557       59       29   0.4915   

   team3PA  team3PM  team3P%  teamFTA  teamFTM  teamFT%  teamORB  teamDRB  \
0       32        8     0.25       20       12   0.6000       18       21   
1       32        8     0.25       20       12   0.6000       18       21   
2       32        8     0.25       20       12   0.6000       18       21   
3       20        7     0.35       22       15   0.6818       18       36   
4       20        7     0.35       22       15   0.6818       18       36   

   teamTRB  teamPTS1  teamPTS2  teamPTS3  teamPTS4  teamPTS5  teamPTS6  \
0       39        24        15        23        22         0         0   
1       39        24        15        23        22         0         0   
2       39        24        15        23        22         0         0   
3       54        31        19        24        20         0         0   
4       54        31        19        24        20         0         0   

   teamPTS7  teamPTS8  teamTREB%  teamASST%  teamTS%  teamEFG%  teamOREB%  \
0         0         0    41.9355    81.2500   0.4251       0.4    33.3333   
1         0         0    41.9355    81.2500   0.4251       0.4    33.3333   
2         0         0    41.9355    81.2500   0.4251       0.4    33.3333   
3         0         0    58.0645    61.1111   0.5300       0.5    46.1538   
4         0         0    58.0645    61.1111   0.5300       0.5    46.1538   

   teamDREB%  teamTO%  teamSTL%  teamBLK%  teamBLKR  teamPPS  teamFIC  \
0    53.8462  11.6279   12.3678   11.2434   17.2414   0.9333    67.25   
1    53.8462  11.6279   12.3678   11.2434   17.2414   0.9333    67.25   
2    53.8462  11.6279   12.3678   11.2434   17.2414   0.9333    67.25   
3    66.6667  19.1466    7.8704    5.6217    8.4746   1.1899    74.00   
4    66.6667  19.1466    7.8704    5.6217    8.4746   1.1899    74.00   

   teamFIC40  teamOrtg  teamDrtg  teamEDiff  teamPlay%   teamAR  teamAST/TO  \
0    56.0417   94.4447  105.6882   -11.2435     0.3765  18.8679      2.0000   
1    56.0417   94.4447  105.6882   -11.2435     0.3765  18.8679      2.0000   
2    56.0417   94.4447  105.6882   -11.2435     0.3765  18.8679      2.0000   
3    61.6667  105.6882   94.4447    11.2435     0.4390  16.7072      1.0476   
4    61.6667  105.6882   94.4447    11.2435     0.4390  16.7072      1.0476   

   teamSTL/TO opptAbbr opptConf    opptDiv opptLoc opptRslt  opptMin  \
0     84.6154      CLE     East    Central    Home      Win      240   
1     84.6154      CLE     East    Central    Home      Win      240   
2     84.6154      CLE     East    Central    Home      Win      240   
3     33.3333      WAS     East  Southeast    Away     Lo

In [50]:
official.to_csv('official_cleaned.csv')

In [51]:
official

gmDate gmTime      offLNm    offFNm teamAbbr teamConf    teamDiv  \
0      2012-10-30  19:00    Brothers      Tony      WAS     East  Southeast   
1      2012-10-30  19:00       Smith   Michael      WAS     East  Southeast   
2      2012-10-30  19:00     Workman  Haywoode      WAS     East  Southeast   
3      2012-10-30  19:00    Brothers      Tony      CLE     East    Central   
4      2012-10-30  19:00       Smith   Michael      CLE     East    Central   
5      2012-10-30  19:00     Workman  Haywoode      CLE     East    Central   
6      2012-10-30  20:00   McCutchen     Monty      BOS     East   Atlantic   
7      2012-10-30  20:00      Wright      Sean      BOS     East   Atlantic   
8      2012-10-30  20:00  Fitzgerald      Kane      BOS     East   Atlantic   
9      2012-10-30  20:00   McCutchen     Monty      MIA     East  Southeast   
10     2012-10-30  20:00      Wright      Sean      MIA     East  Southeast   
11     2012-10-30  20:00  Fitzgerald      Kane      MIA     East  Southeast   
12     2012-10-30  22:30      Foster     Scott      DAL     West  Southwest   
13     2012-10-30  22:30   Zielinski      Gary      DAL     West  Southwest   
14     2012-10-30  22:30       Dalen      Eric      DAL     West  Southwest   
15     2012-10-30  22:30      Foster     Scott      LAL     West    Pacific   
16     2012-10-30  22:30   Zielinski      Gary      LAL     West    Pacific   
17     2012-10-30  22:30       Dalen      Eric      LAL     West    Pacific   
18     2012-10-31  19:00   Salvatore   Bennett      DEN     West  Northwest   
19     2012-10-31  19:00       Brown      Tony      DEN     West  Northwest   
20     2012-10-31  19:00       Lewis      Eric      DEN     West  Northwest   
21     2012-10-31  19:00   Salvatore   Bennett      PHI     East   Atlantic   
22     2012-10-31  19:00       Brown      Tony      PHI     East   Atlantic   
23     2012-10-31  19:00       Lewis      Eric      PHI     East   Atlantic   
24     2012-10-31  19:00     Spooner      Bill      IND     East    Central   
25     2012-10-31  19:00     Guthrie     David      IND     East    Central   
26     2012-10-31  19:00     Lindsay      Mark      IND     East    Central   
27     2012-10-31  19:00     Spooner      Bill      TOR     East   Atlantic   
28     2012-10-31  19:00     Guthrie     David      TOR     East   Atlantic   
29     2012-10-31  19:00     Lindsay      Mark      TOR     East   Atlantic   
...           ...    ...         ...       ...      ...      ...        ...   
44254  2018-04-11  08:00      Wright      Sean      WAS     East  Southeast   
44255  2018-04-11  08:00      Taylor    Dedric      WAS     East  Southeast   
44256  2018-04-11  08:00      Corbin      Sean      WAS     East  Southeast   
44257  2018-04-11  08:00      Wright      Sean      ORL     East  Southeast   
44258  2018-04-11  08:00      Taylor    Dedric      ORL     East  Southeast   
44259  2018-04-11  08:00      Corbin      Sean      ORL     East  Southeast   
44260  2018-04-11  08:00    Williams     James      MIL     East    Central   
44261  2018-04-11  08:00        Lane      Karl      MIL     East    Central   
44262  2018-04-11  08:00       Mauer       Ken      MIL     East    Central   
44263  2018-04-11  08:00    Williams     James      PHI     East   Atlantic   
44264  2018-04-11  08:00        Lane      Karl      PHI     East   Atlantic   
44265  2018-04-11  08:00       Mauer       Ken      PHI     East   Atlantic   
44266  2018-04-11  10:30   Garretson       Ron      LAL     West    Pacific   
44267  2018-04-11  10:30        Mott    Rodney      LAL     West    Pacific   
44268  2018-04-11  10:30  Richardson     Derek      LAL     West    Pacific   
44269  2018-04-11  10:30   Garretson       Ron      LAC     West    Pacific   
44270  2018-04-11  10:30        Mott    Rodney      LAC     West    Pacific   
44271  2018-04-11  10:30  Richardson     Derek      LAC     West    Pacific   
44272  2018-04-11  10:30      Cutler     Kevin      UTA     W

In [52]:
players = pd.read_csv('data/2012-18_playerBoxScore.csv')
standings = pd.read_csv('data/2012-18_standings.csv')

standings.rename(index=str, columns = {"stDate": "gmDate"}, inplace=True)

set(official.columns).difference(set(players.columns))
print(players.shape, standings.shape)
data = players.merge(standings, on = ["gmDate", "teamAbbr"])

print(players.shape, standings.shape, data.shape)
set(data.columns).difference(set(official.columns))

data = data.merge(official, on = ['gmDate', 'teamAbbr'])
print(data.shape)
data

(155713, 51) (29520, 39)
(155713, 51) (29520, 39) (155106, 88)
(465421, 204)


gmDate gmTime_x  seasTyp       playLNm   playFNm teamAbbr  \
0       2012-10-30    19:00  Regular         Price      A.J.      WAS   
1       2012-10-30    19:00  Regular         Price      A.J.      WAS   
2       2012-10-30    19:00  Regular         Price      A.J.      WAS   
3       2012-10-30    19:00  Regular         Ariza    Trevor      WAS   
4       2012-10-30    19:00  Regular         Ariza    Trevor      WAS   
5       2012-10-30    19:00  Regular         Ariza    Trevor      WAS   
6       2012-10-30    19:00  Regular        Okafor     Emeka      WAS   
7       2012-10-30    19:00  Regular        Okafor     Emeka      WAS   
8       2012-10-30    19:00  Regular        Okafor     Emeka      WAS   
9       2012-10-30    19:00  Regular          Beal   Bradley      WAS   
10      2012-10-30    19:00  Regular          Beal   Bradley      WAS   
11      2012-10-30    19:00  Regular          Beal   Bradley      WAS   
12      2012-10-30    19:00  Regular        Booker    Trevor      WAS   
13      2012-10-30    19:00  Regular        Booker    Trevor      WAS   
14      2012-10-30    19:00  Regular        Booker    Trevor      WAS   
15      2012-10-30    19:00  Regular      Crawford    Jordan      WAS   
16      2012-10-30    19:00  Regular      Crawford    Jordan      WAS   
17      2012-10-30    19:00  Regular      Crawford    Jordan      WAS   
18      2012-10-30    19:00  Regular       Webster   Martell      WAS   
19      2012-10-30    19:00  Regular       Webster   Martell      WAS   
20      2012-10-30    19:00  Regular       Webster   Martell      WAS   
21      2012-10-30    19:00  Regular        Vesely       Jan      WAS   
22      2012-10-30    19:00  Regular        Vesely       Jan      WAS   
23      2012-10-30    19:00  Regular        Vesely       Jan      WAS   
24      2012-10-30    19:00  Regular     Singleton     Chris      WAS   
25      2012-10-30    19:00  Regular     Singleton     Chris      WAS   
26      2012-10-30    19:00  Regular     Singleton     Chris      WAS   
27      2012-10-30    19:00  Regular        Barron      Earl      WAS   
28      2012-10-30    19:00  Regular        Barron      Earl      WAS   
29      2012-10-30    19:00  Regular        Barron      Earl      WAS   
...            ...      ...      ...           ...       ...      ...   
465391  2018-04-11    10:30  Regular        Onuaku   Chinanu      HOU   
465392  2018-04-11    10:30  Regular        Onuaku   Chinanu      HOU   
465393  2018-04-11    10:30  Regular        Onuaku   Chinanu      HOU   
465394  2018-04-11    10:30  Regular    Bogdanovic    Bogdan      SAC   
465395  2018-04-11    10:30  Regular    Bogdanovic    Bogdan      SAC   
465396  2018-04-11    10:30  Regular    Bogdanovic    Bogdan      SAC   
465397  2018-04-11    10:30  Regular       Jackson    Justin      SAC   
465398  2018-04-11    10:30  Regular       Jackson    Justin      SAC   
465399  2018-04-11    10:30  Regular       Jackson    Justin      SAC   
465400  2018-04-11    10:30  Regular  Cauley-Stein    Willie      SAC   
465401  2018-04-11    10:30  Regular  Cauley-Stein    Willie      SAC   
465402  2018-04-11    10:30  Regular  Cauley-Stein    Willie      SAC   
465403  2018-04-11    10:30  Regular           Fox  De'Aaron      SAC   
465404  2018-04-11    10:30  Regular           Fox  De'Aaron      SAC   
465405  2018-04-11    10:30  Regular           Fox  De'Aaron      SAC   
465406  2018-04-11    10:30  Regular       Sampson    JaKarr      SAC   
465407  2018-04-11    10:30  Regular       Sampson    JaKarr      SAC   
465408  2018-04-11    10:30  Regular       Sampson    JaKarr      SAC   
465409  2018-04-11    10:30  Regular         Hield   Chavano      SAC   
465410  2018-04-11    10:30  Regular         Hield   Chavano      SAC   
465411  2018-04-11    10:30  Regular         Hield   Chavano      SAC   
465412  2018-04-11    10:30  Regular         Hayes     Nigel      SAC   
465413  2018-04-11    10:30  Regular         Hayes     Nigel      SAC 

In [58]:
(data.memory_usage(deep=True) * 1e-6).sort_values(ascending=False)

playDispNm      33.118789
gmDate          31.183207
playBDate       31.183207
teamDiv_x       30.331211
teamDiv_y       30.331211
opptDiv_x       30.329771
opptDiv_y       30.329771
playLNm         30.011539
seasTyp         29.786944
offLNm1         29.616979
offLNm          29.548174
offLNm2         29.544833
offLNm3         29.479960
playStat        29.297262
playFNm         29.239377
gmTime_y        28.856102
gmTime_x        28.856102
offFNm3         28.752520
offFNm          28.709658
offFNm1         28.699878
offFNm2         28.674178
opptLoc_y       28.390681
teamConf_y      28.390681
opptConf_x      28.390681
opptLoc_x       28.390681
teamLoc_x       28.390681
teamConf_x      28.390681
opptConf_y      28.390681
opptRslt_x      28.158365
opptRslt_y      28.158365
                  ...    
teamFGM          3.723368
teamFGA          3.723368
teamPF           3.723368
poss             3.723368
teamPTS5         3.723368
teamPTS7         3.723368
opptDayOff_y     3.723368
teamPTS8    

In [54]:
data.columns

Index(['gmDate', 'gmTime_x', 'seasTyp', 'playLNm', 'playFNm', 'teamAbbr',
       'teamConf_x', 'teamDiv_x', 'teamLoc_x', 'teamRslt_x',
       ...
       'opptFIC40', 'opptOrtg', 'opptDrtg', 'opptEDiff', 'opptPlay%', 'opptAR',
       'opptAST/TO', 'opptSTL/TO', 'poss', 'pace'],
      dtype='object', length=204)

In [59]:
data.seasTyp.unique()

array(['Regular'], dtype=object)

In [ ]:
data = data.drop(['Regular',],axis=1)

In [63]:
data[data['offLNm1'] == 'Richardson']

gmDate gmTime_x  seasTyp      playLNm    playFNm teamAbbr  \
873     2012-11-02    19:00  Regular       George       Paul      IND   
874     2012-11-02    19:00  Regular       George       Paul      IND   
875     2012-11-02    19:00  Regular       George       Paul      IND   
876     2012-11-02    19:00  Regular         West      David      IND   
877     2012-11-02    19:00  Regular         West      David      IND   
878     2012-11-02    19:00  Regular         West      David      IND   
879     2012-11-02    19:00  Regular         Hill     George      IND   
880     2012-11-02    19:00  Regular         Hill     George      IND   
881     2012-11-02    19:00  Regular         Hill     George      IND   
882     2012-11-02    19:00  Regular        Green     Gerald      IND   
883     2012-11-02    19:00  Regular        Green     Gerald      IND   
884     2012-11-02    19:00  Regular        Green     Gerald      IND   
885     2012-11-02    19:00  Regular      Hibbert        Roy      IND   
886     2012-11-02    19:00  Regular      Hibbert        Roy      IND   
887     2012-11-02    19:00  Regular      Hibbert        Roy      IND   
888     2012-11-02    19:00  Regular   Stephenson      Lance      IND   
889     2012-11-02    19:00  Regular   Stephenson      Lance      IND   
890     2012-11-02    19:00  Regular   Stephenson      Lance      IND   
891     2012-11-02    19:00  Regular      Mahinmi        Ian      IND   
892     2012-11-02    19:00  Regular      Mahinmi        Ian      IND   
893     2012-11-02    19:00  Regular      Mahinmi        Ian      IND   
894     2012-11-02    19:00  Regular     Augustin       D.J.      IND   
895     2012-11-02    19:00  Regular     Augustin       D.J.      IND   
896     2012-11-02    19:00  Regular     Augustin       D.J.      IND   
897     2012-11-02    19:00  Regular   Hansbrough      Tyler      IND   
898     2012-11-02    19:00  Regular   Hansbrough      Tyler      IND   
899     2012-11-02    19:00  Regular   Hansbrough      Tyler      IND   
900     2012-11-02    19:00  Regular        Young        Sam      IND   
901     2012-11-02    19:00  Regular        Young        Sam      IND   
902     2012-11-02    19:00  Regular        Young        Sam      IND   
...            ...      ...      ...          ...        ...      ...   
464740  2018-04-11    08:00  Regular        Hearn     Reggie      DET   
464741  2018-04-11    08:00  Regular        Hearn     Reggie      DET   
464742  2018-04-11    08:00  Regular        Hearn     Reggie      DET   
464743  2018-04-11    08:00  Regular        Payne    Cameron      CHI   
464744  2018-04-11    08:00  Regular        Payne    Cameron      CHI   
464745  2018-04-11    08:00  Regular        Payne    Cameron      CHI   
464746  2018-04-11    08:00  Regular        Nwaba      David      CHI   
464747  2018-04-11    08:00  Regular        Nwaba      David      CHI   
464748  2018-04-11    08:00  Regular        Nwaba      David      CHI   
464749  2018-04-11    08:00  Regular      Felício  Cristiano      CHI   
464750  2018-04-11    08:00  Regular      Felício  Cristiano      CHI   
464751  2018-04-11    08:00  Regular      Felício  Cristiano      CHI   
464752  2018-04-11    08:00  Regular   Kilpatrick       Sean      CHI   
464753  2018-04-11    08:00  Regular   Kilpatrick       Sean      CHI   
464754  2018-04-11    08:00  Regular   Kilpatrick       Sean      CHI   
464755  2018-04-11    08:00  Regular    Markkanen      Lauri      CHI   
464756  2018-04-11    08:00  Regular    Markkanen      Lauri      CHI   
464757  2018-04-11    08:00  Regular    Markkanen      Lauri      CHI   
464758  2018-04-11    08:00  Regular        Grant     Holdyn      CHI   
464759  2018-04-11    08:00  Regular        Grant     Holdyn      CHI   
464760  2018-04-11    08:00  Regular        Grant     Holdyn      CHI   
464761  2018-04-11    08:00  Regular  Arcidiacono       Ryan      CHI   
464762  2018-04-11    08:00  Regular  Arcidiacono       Ryan      CHI 